redshift_sqlalchemy/dialect.py

```diff
@@ -20,6 +24,62 @@ class RedshiftImpl(postgresql.PostgresqlImpl):
         __dialect__ = 'redshift'
 
 
+# Regex for parsing and identity constraint out of adsrc, e.g.:
+#   "identity"(445178, 0, '1,1'::text)
+IDENTITY_RE = re.compile(r"""
+    "identity" \(
+      (?P<current>-?\d+)
+      ,\s
+      (?P<base>-?\d+)
+      ,\s
+      '(?P<seed>-?\d+),(?P<step>-?\d+)'
+      .*
+    \)
+""", re.VERBOSE)
+
+# Regex for foreign key constraints, e.g.:
+#   FOREIGN KEY(col1) REFERENCES othertable (col2)
+# See https://docs.aws.amazon.com/redshift/latest/dg/r_names.html
+# for a definition of valid SQL identifiers.
+FK_RE = re.compile(r"""
+  ^FOREIGN\ KEY \s* \(   # FOREIGN KEY, arbitrary whitespace, literal '('
+    (?P<columns>         # Start a group to capture the referring columns
+      (?:                # Start a non-capturing group
+        \s*              # Arbitrary whitespace
+        [_a-zA-Z][\w$]*  # SQL identifier
```

Comment:
This SQL identifier should be `([_a-zA-Z][\w$]* | "[^"]+")` to support quoted identifiers